drivers/include/sdcard_spi.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2016 Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sdcard_spi SPI SD-Card driver
+ * @ingroup     drivers
+ * @brief       Driver for reading and writing sd-cards via spi interface.
+ * @{
+ *
+ * @file
+ * @brief       Public interface for the sdcard_spi driver.
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef SDCARD_SPI_H
+#define SDCARD_SPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "stdbool.h"
+#include "sdcard_spi_internal.h"
+
+#define SD_HC_BLOCK_SIZE 512
+
+#ifndef NUM_OF_SD_CARDS
+    #define NUM_OF_SD_CARDS 1
+#endif
```

Comment:
I would rather have this driver following the same paradigm as our other device drivers for defining parameters and deducting the the number of defined devices:
- have a default parameter configuration in `drivers/sdcard_spi/include/sdcard_spi_params.h`
- use auto_init (as e.g. `sys/auto_init/netif/auto_init_w5100.c`, of course ommiting the gnrc specific parts)
- define the device descriptor `sdcard_spi_t` and a params struct `sdcard_spi_params_t` in this header file
- have an init function, that takes exactly those two prameters: `int sdcard_init(sdcard_spi_t *dev, sdcard_spi_params_t *params);`